test/integration/framework/utils.go

```diff
@@ -0,0 +1,358 @@
+	var runningPodWithDesiredLabels *corev1.Pod
+	set := labels.Set(labelsMap)
+
+	var podList *corev1.PodList
+	var err error
+
+	switch seedOrShoot {
+	case "seed":
+		podList, err = o.K8sSeedClient.ListPods(o.Shoot.SeedNamespace, metav1.ListOptions{
+			LabelSelector: set.AsSelector().String(),
+		})
+		if err != nil {
+			return nil, err
+		}
+	case "shoot":
+		podList, err = o.K8sShootClient.ListPods(namespace, metav1.ListOptions{
+			LabelSelector: set.AsSelector().String(),
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Fetch the first running pod of the Kube-apiserver
+	if len(podList.Items) > 0 {
+		pod := podList.Items[0]
+		o.ShootGardenerTest.Logger.Infof("%s:%s", pod.Namespace, pod.Name)
+		runningPodWithDesiredLabels = &pod
+	}
+	return runningPodWithDesiredLabels, nil
+
+}
+
+// GetAdminPassword gets the admin password for authenticating agains the api
+func (o *GardenerTestOperation) getAdminPassword() (*string, error) {
+	password, err := getPasswordFromSecret(o.K8sSeedClient, o.Shoot.SeedNamespace, "kubecfg")
+	if err != nil {
+		return nil, err
+	}
+	return &password, nil
+}
+
+func (s *ShootGardenerTest) mergePatch(oldShoot, newShoot []byte) error {
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldShoot, newShoot, v1beta1.Shoot{})
+	if err != nil {
+		return fmt.Errorf("failed to patch bytes")
+	}
+
+	_, err = s.K8sGardenClient.Garden().GardenV1beta1().Shoots(s.Shoot.Namespace).Patch(s.Shoot.Name, types.StrategicMergePatchType, patchBytes)
+	if err != nil {
+		return fmt.Errorf("failed to update shoot with new value: %s", err.Error())
+	}
+
+	return nil
+}
+
+func shootCreationCompleted(newStatus *v1beta1.ShootStatus) bool {
+	validConditionMatrix := true
+	if len(newStatus.Conditions) > 0 {
+		for _, condition := range newStatus.Conditions {
+			validConditionMatrix = validConditionMatrix && (condition.Status == "True")
+		}
+		if newStatus.LastOperation != nil {
+			if newStatus.LastOperation.Type == v1beta1.ShootLastOperationTypeCreate ||
+				newStatus.LastOperation.Type == v1beta1.ShootLastOperationTypeReconcile {
+				validConditionMatrix = validConditionMatrix && (newStatus.LastOperation.State == v1beta1.ShootLastOperationStateSucceeded)
+			}
+		}
+	} else {
+		validConditionMatrix = false
+	}
+
+	return validConditionMatrix
+}
+
+func getPasswordFromSecret(k8sClient kubernetes.Interface, namespace, secretName string) (string, error) {
+	secret, err := k8sClient.GetSecret(namespace, secretName)
+	if err != nil {
+		return "", err
+	}
+
+	if _, ok := secret.Data["password"]; ok {
+		return string(secret.Data["password"]), nil
+	}
+
+	return "", nil
+
+}
+func getKubeconfigFromSecret(k8sClient kubernetes.Interface, namespace, secretName string) (string, error) {
+	secret, err := k8sClient.GetSecret(namespace, secretName)
+	if err != nil {
+		return "", err
+	}
+
+	if _, ok := secret.Data["kubeconfig"]; ok {
```

Comment:
Again, use the value directly and use constants. As this also makes sense for our `client` code, I'd even suggest factoring this out there and reusing it here.